### DISCUSSION TITLE
How to redirect to Same url after Google Sign in ?

### DISCUSSION THREAD

**Discussion by roker15** [supabase/supabase#27658] [Questions]
Currently You have to redirect to `app/auth/callback/route.ts`  in supabase-nextjs project to handle code exchange. There is no way so that i can redirect to same URL where sign IN began. Because in redirectTo i always need to pass `redirectTo: {window.location.origin}/auth/callback` and not the current page url. 

Supabse page also has no clew... https://supabase.com/docs/guides/auth/social-login/auth-google?queryGroups=environment&environment=client

**Comment by j4w8n** [comment]
At the end of the redirectTo URL, add `?next=/path/you/want` and the callback route should redirect the user after exchanging the code for a session.

**Reply by roker15** [reply]
> At the end of the redirectTo URL, add `?next=/path/you/want` and the callback route should redirect the user after exchanging the code for a session.

What about query params? Many times query params required to display full page what user want to see actually....

**Reply by j4w8n** [reply]
> > At the end of the redirectTo URL, add `?next=/path/you/want` and the callback route should redirect the user after exchanging the code for a session.
> 
> What about query params? Many times query params required to display full page what user want to see actually....

I believe you can add those before the change I suggested, but I have no experience with that.

**Reply by j4w8n** [reply]
Unless you're talking about query params for the redirect route. In which case you should be able to add those after your `next` path.

**Reply by roker15** [reply]
> Unless you're talking about query params for the redirect route. In which case you should be able to add those after your `next` path.

that should be sent while sign in begans at client, query params are dynamic. I can not add each time manually when it hits route `app/auth/callback/route.ts`
